src/gp/inc/GPMSAOptions.h

```diff
@@ -86,6 +87,104 @@ class GPMSAOptions
   //  Currently the only supported value is 1.0
   double m_emulatorBasisVarianceToCapture;
 
+  //! Do automatic normalization, using minimum and maximum values in
+  //  the supplied data, for uncertain parameter i.
+  //
+  //  Normalized values will range from 0 to 1.
+  void set_autoscale_minmax_uncertain_parameter(unsigned int i);
+
+  //! Do automatic normalization, using minimum and maximum values in
+  //  the supplied data, for scenario parameter i.
+  //
+  //  Normalized values will range from 0 to 1.
+  void set_autoscale_minmax_scenario_parameter(unsigned int i);
+
+  //! Do automatic normalization, using minimum and maximum values in
+  //  the supplied data, for all input uncertain parameters, all input
+  //  scenario parameters, and all output values.
+  //
+  //  Normalized values will range from 0 to 1.
+  void set_autoscale_minmax();
+
+  //! Do automatic normalization, using mean and variance of the
+  //  supplied data, for uncertain parameter i.
+  //
+  //  Normalized values will have mean 0 and standard deviation 1.
+  void set_autoscale_meanvar_uncertain_parameter(unsigned int i);
+
+  //! Do automatic normalization, using mean and variance of the
+  //  supplied data, for scenario parameter i.
+  //
+  //  Normalized values will have mean 0 and standard deviation 1.
+  void set_autoscale_meanvar_scenario_parameter(unsigned int i);
+
+  //! Do automatic normalization, using mean and variance of the
+  //  supplied data, for all input uncertain parameters, all input
+  //  scenario parameters, and all output values.
+  //
+  //  Normalized values will have mean 0 and standard deviation 1.
+  void set_autoscale_meanvar();
+
+  //! Set a value, for uncertain parameter i in simulation inputs, of
+  //  the physical parameter range (range_min, range_max) which should
+  //  be rescaled to (0,1)
```

Comment:
It took me a minute to realise why we need to allow a general range.  It's because the user may want to specify a range besides the min/max of the supplied data.